jekyll/_cci2/workflows.adoc

```diff
@@ -107,6 +107,52 @@ Define job dependencies using the `requires` key. A job must wait until all upst
 
 See the link:https://github.com/CircleCI-Public/circleci-demo-workflows/blob/sequential-branch-filter/.circleci/config.yml[Sample Sequential Workflow config] for a full example.
 
+[#flexible-job-dependency]
+==== Flexible job dependency
```

Comment:
I think it would be good to move this section to the end of this sub section, so an h3 after the fan in-out section?

![Screenshot 2025-01-08 at 13 41 41](https://github.com/user-attachments/assets/3bb0aef1-8cc9-4b80-8b11-8239cd0f41e2)